autofit/graphical/expectation_propagation/ep_mean_field.py

```diff
@@ -215,7 +215,6 @@ def factor_approximation(self, factor: Factor) -> FactorApproximation:
             for v in factor_dist.all_variables
         }).prod(
             *factor_mean_field.values(),
-            default=factor_dist,
```

Comment:
This causes incorrect initialisation of the cavity dist if a variable is used by only one factor.

Reply:
It's quite common in our usage for variables to only be used by a single factor. Many factors have 'dangling' variables which are not shared

Reply:
Do you mean you've fixed that issue or newly introduced it?

Reply:
I think I fixed it. 

The way that you solved the dangling variable problem was causing variance shrinkage in the case of dangling variables.

Reply:
This way the dangling variable has no cavity distribution associated with it in the `FactorApproximation` which is fine for a lot of factor optimisations.

If you require access to the cavity distribution as a prior then you shouldn't have a dangling variable. (The prior should be present as a second factor connected to the variable).

Reply:
Actually yeah I guess every variable at least has a PriorFactor attached and so would be shared.

Reply:
A test for this happening would be to run a factor optimisation multiple times in a row for the same factor, if the variance shrinks each step, then you're doing something like this.